### PR TITLE
feat: integrate ipfs-repo-migrations tool

### PR DESCRIPTION
**Pull request by AuHau**
Resolves: https://github.com/ipfs/js-ipfs/issues/1115

Supersedes: https://github.com/ipfs/js-ipfs/pull/2044

Depends on:

* [x] https://github.com/ipfs/js-ipfs-repo/pull/202

### CODE REVIEW

**Comment by AuHau** [comment]
@daviddias should it be part of `js-ipfs` or `interface-js-ipfs-core`?

**Comment by AuHau** [comment]
Hmmm that is a good point, but `daemon` is a bit too late, it would have to be part of the `js-ipfs` root command because you can interact with the repo without having daemon on and hence the migrations will run already during for example `js-ipfs key list`

**Comment by AuHau** [comment]
@alanshaw ping ^

**Comment by alanshaw** [comment]
I just meant we can leave it for now.

**Comment by alanshaw** [comment]
@AuHau and I spoke about this in IRC and decided to default to true for auto migrations in core but when run via the CLI or as a daemon it would be set to false, allowing the user to backup their datastore before migrating. A new global option called --migrate will set it to true.

**Comment by AuHau** [comment]
So it is ready for review. Unfortunately, I did not manage to add any tests. I am now pressed to put time somewhere else, so I don't think I will be able to work on tests anytime soon if they will be needed...

**Comment by alanshaw** [comment]
No worries, I'll pick this up from here - thanks for all your work on this @AuHau ❤️

File: README.md [hunk 1]
```diff
@@ -284,6 +284,18 @@ Example:
 const node = await IPFS.create({ repo: '/var/ipfs/data' })
 ```
 
+##### `options.repoAutoMigrate`
+
+| Type | Default |
+|------|---------|
+| boolean | `true` |
+
+`js-ipfs` comes bundled with tool that automatically migrate the version of your IPFS repository when new version is available.
```
Comment (alanshaw) [inline README.md]:
```suggestion
`js-ipfs` comes bundled with a tool that automatically migrates your IPFS repository when a new version is available.
```

File: README.md [hunk 2]
```diff
@@ -284,6 +284,18 @@ Example:
...
+
+**For tools that build on top of `js-ipfs` and run mainly in the browser environment, be aware that disabling automatic
```
Comment (alanshaw) [inline README.md]:
```suggestion
**For apps that build on top of `js-ipfs` and run in the browser environment, be aware that disabling automatic
```

File: src/cli/parser.js [hunk 3]
```diff
@@ -19,6 +19,11 @@ const parser = yargs
     type: 'string',
     default: ''
   })
+  .option('migrate', {
+    desc: 'Allows automatic migrations',
```
Comment (alanshaw) [inline src/cli/parser.js]:
```suggestion
    desc: 'Enable/disable automatic repo migrations',
```

File: src/cli/utils.js [hunk 4]
```diff
@@ -60,7 +61,11 @@ exports.getIPFS = (argv, callback) => {
   })
 
   node.on('error', (err) => {
-    throw err
+    if (err.code === 'ERR_INVALID_REPO_VERSION') {
+      err.message = 'Incompatible repo version. Migration needed. Pass --migrate for automatic migration'
+    }
+
+    callback(null, node, cleanup)
```
Comment (alanshaw) [inline src/cli/utils.js]:
```suggestion
    callback(err)
```

**Pull request merged**